From 544cbcb52676d3dff8810cbf313ba3e8f5d20e1d Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Sat, 17 Jun 2017 01:16:20 +0000
Subject: [PATCH] DOCKER-1070 Don't recreate "sdc_docker" rule when it's been
 disabled Reviewed by: Todd Whiteman <todd.whiteman@joyent.com> Approved by:
 Todd Whiteman <todd.whiteman@joyent.com>

---
 lib/backends/sdc/containers.js | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/lib/backends/sdc/containers.js b/lib/backends/sdc/containers.js
index a99acac..516d192 100644
--- a/lib/backends/sdc/containers.js
+++ b/lib/backends/sdc/containers.js
@@ -265,10 +265,6 @@ function addExposeFirewallRules(opts, payload, callback) {
                 continue;
             }
 
-            if (!rules[r].enabled) {
-                continue;
-            }
-
             if (parsed.action !== 'allow') {
                 continue;
             }
@@ -286,14 +282,20 @@ function addExposeFirewallRules(opts, payload, callback) {
                 tcpRuleFound = true;
                 opts.log.debug({ rule: rules[r] },
                     'TCP docker expose rule found: not adding');
-                addRulesToPayload(payload, rules[r]);
+
+                if (parsed.enabled) {
+                    addRulesToPayload(payload, rules[r]);
+                }
             }
 
             if (parsed.protocol === 'udp' && !udpRuleFound) {
                 udpRuleFound = true;
                 opts.log.debug({ rule: rules[r] },
                     'UDP docker expose rule found: not adding');
-                addRulesToPayload(payload, rules[r]);
+
+                if (parsed.enabled) {
+                    addRulesToPayload(payload, rules[r]);
+                }
             }
         }
 
-- 
2.21.0

